### ISSUE TITLE
Weights are updated twice on each task?

### ISSUE THREAD

**Issue by ztf-ucas** [jik0730/Meta-SGD-pytorch#2]
I'm sorry for disturbing you again .  
I was confused about the train_updates operation (as follows) when I compare the code with the pseudo-code in Meta-SGD paper where weights are updated only once for each task but twice  here.

    adapted_state_dict = model.cloned_state_dict()
    adapted_params = OrderedDict()
    for (key, val), grad in zip(model.named_parameters(), grads):
        # NOTE Here Meta-SGD is different from naive MAML
        task_lr = model.task_lr[key]
        adapted_params[key] = val - task_lr * grad
        adapted_state_dict[key] = adapted_params[key]

    # NOTE In Meta-SGD paper, num_train_updates=1 is enough
    for _ in range(1, num_train_updates):
        Y_sup_hat = model(X_sup, adapted_state_dict)
        loss = loss_fn(Y_sup_hat, Y_sup)
        zero_grad(adapted_params.values())
        # optimizer.zero_grad()
        # loss.backward(create_graph=True)
        grads = torch.autograd.grad(
            loss, adapted_params.values(), create_graph=True)
        for (key, val), grad in zip(adapted_params.items(), grads):
            task_lr = model.task_lr[key]
            adapted_params[key] = val - task_lr * grad
            adapted_state_dict[key] = adapted_params[key]

    return adapted_state_dict

**Comment by jik0730** [comment]
Hi again :) Feel free to ask questions about codes!
I acknowledge that the parts you mentioned are a bit confusing. As you said, train_updates operation should be set to be only one as in Meta-SGD paper. By that reason, "num_train_update" and "num_eval_updates" in params.json should be set to be one to follow the settings in original paper. Thus, the parts below the line 9 (NOTE In Meta-SGD ...) are not needed for training. Also since num_train_updates=1, as I mentioned, for loop in the parts does not even operate, which means the functionality is correct (where the weights are updated only once).
I will update the codes soon and thanks for your comment.

**Comment by ztf-ucas** [comment]
I got it！ Thank you very much！
Your code is a good example for me to learn pytorch.
